### ISSUE TITLE
[PyQt6 6.6.1, macOS 14] Hard crash when toggling to 3D mode, RGB image, intermittent

### ISSUE THREAD

**Issue by psobolewskiPhD** [napari/napari#6550]
### 🐛 Bug Report

Using `main`, 0.4.18 and 0.4.19rc3 when installing with `pip install "napari[pyqt6_experimental]"` things seem fine until I toggle an image from 2D to 3D. This results in a hard crash:
`libc++abi: terminating due to uncaught exception of type NSException`
Full trace:
<details>

```
*** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArrayM objectAtIndexedSubscript:]: index 0 beyond bounds for empty array'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000018d4ce800 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x000000018cfc5eb4 objc_exception_throw + 60
	2   CoreFoundation                      0x000000018d444fc0 -[__NSCFString hasSuffix:] + 0
	3   libqcocoa.dylib                     0x000000016aa8e044 _ZN20QCocoaSystemTrayIcon13emitActivatedEv + 274796
	4   libqcocoa.dylib                     0x000000016aa8e0c8 _ZN20QCocoaSystemTrayIcon13emitActivatedEv + 274928
	5   libqcocoa.dylib                     0x000000016aa8c688 _ZN20QCocoaSystemTrayIcon13emitActivatedEv + 268208
	6   QtWidgets                           0x000000013dd391a8 _ZN9QListView16selectionChangedERK14QItemSelectionS2_ + 204
	7   QtCore                              0x0000000139eb748c _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3808
	8   QtCore                              0x000000013a07de44 _ZN19QItemSelectionModel20emitSelectionChangedERK14QItemSelectionS2_ + 216
	9   QtCore                              0x000000013a07da2c _ZN19QItemSelectionModel6selectERK14QItemSelection6QFlagsINS_13SelectionFlagEE + 636
	10  QtCore                              0x000000013a07d70c _ZN19QItemSelectionModel6selectERK11QModelIndex6QFlagsINS_13SelectionFlagEE + 68
	11  QtCore                              0x000000013a07e8cc _ZN19QItemSelectionModel15setCurrentIndexERK11QModelIndex6QFlagsINS_13SelectionFlagEE + 260
	12  QtWidgets                           0x000000013db97338 _ZNK9QComboBox5countEv + 828
	13  QtWidgets                           0x000000013db9a8b4 _ZN9QComboBoxC1ER16QComboBoxPrivateP7QWidget + 2412
	14  QtCore                              0x0000000139eb7224 _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3192
	15  QtCore                              0x000000013a066944 _ZN18QAbstractItemModel13endInsertRowsEv + 200
	16  QtGui                               0x000000013d07f1a8 _ZN18QStandardItemModel11itemChangedEP13QStandardItem + 844
	17  QtWidgets                           0x000000013db9de38 _ZN9QComboBox11insertItemsEiRK5QListI7QStringE + 340
	18  QtWidgets.abi3.so                   0x000000013c9529bc _ZL23meth_QComboBox_addItemsP7_objectS0_ + 136
	19  python3.11                          0x000000010015334c cfunction_call + 284
	20  python3.11                          0x00000001000fc1fc _PyObject_MakeTpCall + 332
	21  python3.11                          0x00000001001fca58 _PyEval_EvalFrameDefault + 47160
	22  python3.11                          0x0000000100200c1c _PyEval_Vector + 184
	23  python3.11                          0x00000001000fcb04 PyObject_Vectorcall + 76
	24  python3.11                          0x000000010010ae48 property_descr_set + 92
	25  python3.11                          0x000000010015ab0c _PyObject_GenericSetAttrWithDict + 260
	26  python3.11                          0x0000000100159b38 PyObject_SetAttr + 148
	27  python3.11                          0x00000001001f5480 _PyEval_EvalFrameDefault + 16992
	28  python3.11                          0x0000000100200c1c _PyEval_Vector + 184
	29  python3.11                          0x00000001000fc3c4 _PyObject_FastCallDictTstate + 156
	30  python3.11                          0x00000001000fd2f8 _PyObject_Call_Prepend + 176
	31  python3.11                          0x00000001001764e0 slot_tp_call + 172
	32  python3.11                          0x00000001000fc1fc _PyObject_MakeTpCall + 332
	33  python3.11                          0x00000001001fca58 _PyEval_EvalFrameDefault + 47160
	34  python3.11                          0x0000000100200c1c _PyEval_Vector + 184
	35  python3.11                          0x00000001000ffcb0 method_vectorcall + 172
	36  python3.11                          0x00000001000fcb04 PyObject_Vectorcall + 76
	37  python3.11                          0x00000001001fca58 _PyEval_EvalFrameDefault + 47160
	38  python3.11                          0x0000000100200c1c _PyEval_Vector + 184
	39  python3.11                          0x000000010017bf14 vectorcall_method + 228
	40  python3.11                          0x0000000100176970 slot_tp_setattro + 60
	41  python3.11                          0x0000000100159b38 PyObject_SetAttr + 148
	42  python3.11                          0x00000001001f5480 _PyEval_EvalFrameDefault + 16992
	43  python3.11                          0x0000000100200c1c _PyEval_Vector + 184
	44  python3.11                          0x00000001000fc970 _PyVectorcall_Call + 132
	45  python3.11                          0x00000001001fe978 _PyEval_EvalFrameDefault + 55128
	46  python3.11                          0x0000000100200c1c _PyEval_Vector + 184
	47  QtCore.abi3.so                      0x0000000139ba6418 _ZNK8PyQtSlot4callEP7_objectS1_ + 156
	48  QtCore.abi3.so                      0x0000000139ba622c _ZNK8PyQtSlot6invokeEPPvP7_objectS0_b + 296
	49  QtCore.abi3.so                      0x0000000139ba7200 _ZN13PyQtSlotProxy7unislotEPPv + 92
	50  QtCore.abi3.so                      0x0000000139ba7170 _ZN13PyQtSlotProxy11qt_metacallEN11QMetaObject4CallEiPPv + 68
	51  QtCore                              0x0000000139eb71f4 _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3144
	52  QtWidgets                           0x000000013db80ec8 _ZNK15QAbstractButton11isCheckableEv + 816
	53  QtWidgets                           0x000000013db80d48 _ZNK15QAbstractButton11isCheckableEv + 432
	54  QtWidgets                           0x000000013db81c48 _ZN15QAbstractButton17mouseReleaseEventEP11QMouseEvent + 172
	55  QtWidgets.abi3.so                   0x000000013c8399ec _ZN14sipQPushButton17mouseReleaseEventEP11QMouseEvent + 148
	56  QtWidgets                           0x000000013dacd8b8 _ZN7QWidget5eventEP6QEvent + 132
	57  QtWidgets                           0x000000013dc2eb24 _ZN11QPushButton5eventEP6QEvent + 204
	58  QtWidgets.abi3.so                   0x000000013c83a41c _ZN14sipQPushButton5eventEP6QEvent + 224
	59  QtWidgets                           0x000000013da84594 _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent + 272
	60  QtWidgets                           0x000000013da865c0 _ZN12QApplication6notifyEP7QObjectP6QEvent + 5072
	61  QtWidgets.abi3.so                   0x000000013c85b224 _ZN15sipQApplication6notifyEP7QObjectP6QEvent + 248
	62  QtCore                              0x0000000139e6d474 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent + 292
	63  QtWidgets                           0x000000013da84b34 _ZN19QApplicationPrivate14sendMouseEventEP7QWidgetP11QMouseEventS1_S1_PS1_R8QPointerIS0_Ebb + 808
	64  QtWidgets                           0x000000013dae2860 _ZNK21QWidgetRepaintManager3rhiEv + 11080
	65  QtWidgets                           0x000000013dae1a6c _ZNK21QWidgetRepaintManager3rhiEv + 7508
	66  QtWidgets                           0x000000013da84594 _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent + 272
	67  QtWidgets                           0x000000013da853e8 _ZN12QApplication6notifyEP7QObjectP6QEvent + 504
	68  QtWidgets.abi3.so                   0x000000013c85b224 _ZN15sipQApplication6notifyEP7QObjectP6QEvent + 248
	69  QtCore                              0x0000000139e6d474 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent + 292
	70  QtGui                               0x000000013cddb9ec _ZN22QGuiApplicationPrivate17processMouseEventEPN29QWindowSystemInterfacePrivate10MouseEventE + 1740
	71  QtGui                               0x000000013ce37674 _ZN22QWindowSystemInterface22sendWindowSystemEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE + 408
	72  libqcocoa.dylib                     0x000000016aa2fd74 qt_plugin_instance + 58768
	73  CoreFoundation                      0x000000018d459cfc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
	74  CoreFoundation                      0x000000018d459c90 __CFRunLoopDoSource0 + 176
	75  CoreFoundation                      0x000000018d459a00 __CFRunLoopDoSources0 + 244
	76  CoreFoundation                      0x000000018d4585f0 __CFRunLoopRun + 828
	77  CoreFoundation                      0x000000018d457c5c CFRun
libc++abi: terminating due to uncaught exception of type NSException
zsh: abort      napari
```

</details>

Note that 6.5.3 is fine. `pip install PyQt6-Qt6==6.6.0 PyQt6==6.6.0` recovers normal behavior.

### 💡 Steps to Reproduce

1. make a Python env (3.11) and install with `pip install "napari[pyqt6_experimental]"`
2. Open Sample Blobs 3D
3. Toggle 3D viewer mode

### 💡 Expected Behavior

no crash

### 🌎 Environment

napari: 0.4.18
Platform: macOS-14.1.2-arm64-arm-64bit
System: MacOS 14.1.2
Python: 3.11.7 | packaged by conda-forge | (main, Dec 15 2023, 08:41:36) [Clang 16.0.6 ]
Qt: 6.6.1
PyQt6: 
NumPy: 1.26.2
SciPy: 1.11.4
Dask: 2023.12.1
VisPy: 0.12.2
magicgui: 0.8.1
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.4
npe2: 0.7.3

OpenGL:
  - GL version:  2.1 Metal - 88
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 2056x1329, scale 2.0
  - screen 2: resolution 2560x1440, scale 1.0

Settings path:
  - /Users/sobolp/Library/Application Support/napari/napari-418_d279d6cf5d4193876ed97858e757fe322037331b/settings.yaml
Plugins:
  - napari: 0.4.18 (77 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-svg: 0.1.10 (2 contributions)

### 💡 Additional Context

As noted above, this happens with main and 0.4.19, so it's triggered by a change (bug?) introduced by PyQt6 6.6.1

**Comment by psobolewskiPhD** [comment]
Note on my personal mac still on Ventura 13.3.1 I cannot replicate this.
```
napari: 0.4.18
Platform: macOS-13.3.1-arm64-arm-64bit
System: MacOS 13.3.1
Python: 3.11.7 | packaged by conda-forge | (main, Dec 15 2023, 08:41:36) [Clang 16.0.6 ]
Qt: 6.6.1
PyQt6: 
NumPy: 1.26.2
SciPy: 1.11.4
Dask: 2023.12.1
VisPy: 0.12.2
magicgui: 0.8.1
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.4
npe2: 0.7.3

OpenGL:
  - GL version:  2.1 Metal - 83.1
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1680x1050, scale 2.0

Settings path:
  - /Users/piotrsobolewski/Library/Application Support/napari/napari-pqt6_2adb881c24ecfed7e11b6818551f569bc76ce2c0/settings.yaml
Plugins:
  - napari: 0.4.18 (77 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-svg: 0.1.10 (2 contributions)
  
```

**Comment by Czaki** [comment]
I was about to ask you to try on `PySide6`, but... 

It looks like Qt6 bug itself. Stack trace reports that bug happens when an item is added to CombBox

**Comment by Czaki** [comment]
https://bugreports.qt.io/browse/QTBUG-118585?filter=-4&jql=text%20~%20%22index%200%20beyond%20bounds%20for%20empty%20array%22%20order%20by%20created%20DESC

It looks like it is fixed upstream and you need to wait on new release of Qt and PyQt6

**Comment by psobolewskiPhD** [comment]
Is anyone else able to reproduce this? Should we !=6.6.1?

**Comment by Czaki** [comment]
We then need to block it only on macOS. 
Based on comments from linked qt issue, the problem is in the macOS accessibility API.

**Comment by lukasz-migas** [comment]
Yes, I get the same error but when opening the plugins dialog via the menu.

MacOS 13.5.2
python 3.10

Stacktrace:
```
Application Specific Backtrace 0:
0   CoreFoundation                      0x000000018543b154 __exceptionPreprocess + 176
1   libobjc.A.dylib                     0x0000000184f5a4d4 objc_exception_throw + 60
2   CoreFoundation                      0x00000001855259b8 -[__NSCFString characterAtIndex:].cold.1 + 0
3   CoreFoundation                      0x00000001853aebb8 -[__NSArrayM objectAtIndexedSubscript:] + 188
4   libqcocoa.dylib                     0x000000028a14e044 _ZN20QCocoaSystemTrayIcon13emitActivatedEv + 274796
5   libqcocoa.dylib                     0x000000028a14e0c8 _ZN20QCocoaSystemTrayIcon13emitActivatedEv + 274928
6   libqcocoa.dylib                     0x000000028a14c688 _ZN20QCocoaSystemTrayIcon13emitActivatedEv + 268208
7   QtWidgets                           0x000000017004d1a8 _ZN9QListView16selectionChangedERK14QItemSelectionS2_ + 204
8   QtCore                              0x000000016a54f48c _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3808
9   QtCore                              0x000000016a715e44 _ZN19QItemSelectionModel20emitSelectionChangedERK14QItemSelectionS2_ + 216
10  QtCore                              0x000000016a715a2c _ZN19QItemSelectionModel6selectERK14QItemSelection6QFlagsINS_13SelectionFlagEE + 636
11  QtCore                              0x000000016a71570c _ZN19QItemSelectionModel6selectERK11QModelIndex6QFlagsINS_13SelectionFlagEE + 68
12  QtCore                              0x000000016a7168cc _ZN19QItemSelectionModel15setCurrentIndexERK11QModelIndex6QFlagsINS_13SelectionFlagEE + 260
13  QtWidgets                           0x000000016feab338 _ZNK9QComboBox5countEv + 828
14  QtWidgets                           0x000000016feae8b4 _ZN9QComboBoxC1ER16QComboBoxPrivateP7QWidget + 2412
15  QtCore                              0x000000016a54f224 _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3192
16  QtCore                              0x000000016a6fe944 _ZN18QAbstractItemModel13endInsertRowsEv + 200
17  QtGui                               0x000000016f3937a4 _ZN18QStandardItemModel11itemChangedEP13QStandardItem + 2376
18  QtWidgets                           0x000000016feb1910 _ZN9QComboBox10insertItemEiRK5QIconRK7QStringRK8QVariant + 368
19  QtWidgets.abi3.so                   0x000000016eb5e7a0 _ZL22meth_QComboBox_addItemP7_objectS0_S0_ + 256
20  python3.10                          0x00000001041d6b3c cfunction_call + 80
21  python3.10                          0x000000010418359c _PyObject_MakeTpCall + 612
22  python3.10                          0x00000001042755d8 call_function + 676
23  python3.10                          0x0000000104270e60 _PyEval_EvalFrameDefault + 24940
24  python3.10                          0x000000010426a364 _PyEval_Vector + 2036
25  python3.10                          0x0000000104186c64 method_vectorcall + 172
26  python3.10                          0x0000000104275540 call_function + 524
27  python3.10                          0x0000000104270e60 _PyEval_EvalFrameDefault + 24940
28  python3.10                          0x000000010426a364 _PyEval_Vector + 2036
29  python3.10                          0x00000001041837ac _PyObject_FastCallDictTstate + 156
30  python3.10                          0x0000000104184454 _PyObject_Call_Prepend + 176
31  python3.10                          0x00000001041fb394 slot_tp_init + 116
32  python3.10                          0x00000001041f3d64 type_call + 456
33  python3.10                          0x000000010418359c _PyObject_MakeTpCall + 612
34  python3.10                          0x00000001042755d8 call_function + 676
35  python3.10                          0x0000000104271430 _PyEval_EvalFrameDefault + 26428
36  python3.10                          0x000000010426a364 _PyEval_Vector + 2036
37  python3.10                          0x0000000104186c64 method_vectorcall + 172
38  python3.10                          0x0000000104275540 call_function + 524
39  python3.10                          0x0000000104270e60 _PyEval_EvalFrameDefault + 24940
40  python3.10                          0x000000010426a364 _PyEval_Vector + 2036
41  python3.10                          0x0000000104186d10 method_vectorcall + 344
42  QtCore.abi3.so                      0x000000016a23e3c0 _ZNK8PyQtSlot4callEP7_objectS1_ + 68
43  QtCore.abi3.so                      0x000000016a23e22c _ZNK8PyQtSlot6invokeEPPvP7_objectS0_b + 296
44  QtCore.abi3.so                      0x000000016a23f200 _ZN13PyQtSlotProxy7unislotEPPv + 92
45  QtCore.abi3.so                      0x000000016a23f170 _ZN13PyQtSlotProxy11qt_metacallEN11QMetaObject4CallEiPPv + 68
46  QtCore                              0x000000016a54f1f4 _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3144
47  QtCore                              0x000000016a561f08 _ZN6QTimer10timerEventEP11QTimerEvent + 208
48  QtCore.abi3.so                      0x000000016a1190e4 _ZN9sipQTimer10timerEventEP11QTimerEvent + 148
49  QtCore                              0x000000016a547ccc _ZN7QObject5eventEP6QEvent + 88
50  QtCore.abi3.so                      0x000000016a118f30 _ZN9sipQTimer5eventEP6QEvent + 220
51  QtWidgets                           0x000000016fd98594 _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent + 272
52  QtWidgets                           0x000000016fd993e8 _ZN12QApplication6notifyEP7QObjectP6QEvent + 504
53  QtWidgets.abi3.so                   0x000000016ea67224 _ZN15sipQApplication6notifyEP7QObjectP6QEvent + 248
54  QtCore                              0x000000016a505474 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent + 292
55  QtCore                              0x000000016a6765c8 _ZN14QTimerInfoList14activateTimersEv + 904
56  libqcocoa.dylib                     0x000000028a0ecc8c qt_plugin_instance + 46248
57  CoreFoundation                      0x00000001853c263c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
58  CoreFoundation                      0x00000001853c25d0 __CFRunLoopDoSource0 + 176
59  CoreFoundation                      0x00000001853c2340 __CFRunLoopDoSources0 + 244
60  CoreFoundation                      0x00000001853c0f48 __CFRunLoopRun + 828
61  CoreFoundation                      0x00000001853c04b8 CFRunLoopRunSpecific + 612
62  HIToolbox                           0x000000018ec12df0 RunCurrentEventLoopInMode + 292
63  HIToolbox                           0x000000018ec12c2c ReceiveNextEventCommon + 648
64  HIToolbox                           0x000000018ec12984 _BlockUntilNextEventMatchingListInModeWithFilter + 76
65  AppKit                              0x00000001885e797c _DPSNextEvent + 636
66  AppKit                              0x00000001885e6b18 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716
67  AppKit                              0x000000018881fbd0 -[NSApplication _doModalLoop:peek:] + 216
68  AppKit                              0x0000000188964844 __33-[NSApplication runModalSession:]_block_invoke_2 + 88
69  AppKit                              0x00000001889647d0 __33-[NSApplication runModalSession:]_block_invoke + 112
70  AppKit                              0x000000018881e364 _NSTryRunModal + 100
71  AppKit                              0x00000001889646a0 -[NSApplication runModalSession:] + 148
72  libqcocoa.dylib                     0x000000028a0ed854 qt_plugin_instance + 49264
73  QtCore                              0x000000016a50f160 _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE + 540
74  QtWidgets                           0x000000016ffb03d8 _ZN7QDialog4execEv + 476
75  QtWidgets.abi3.so                   0x000000016eb664ec _ZL17meth_QDialog_execP7_objectS0_ + 208
76  python3.10                          0x00000001041d6bbc cfunction_call + 208
77  python3.10                          0x0000000104183fb0 _PyObject_Call + 148
78  python3.10                          0x0000000104271658 _PyEval_EvalFrameDefault + 26980
79  python3.10                          0x000000010426a364 _PyEval_Vector + 2036
80  python3.10                          0x0000000104186c64 method_vectorcall + 172
81  python3.10                          0x0000000104275540 call_function + 524
82  python3.10                          0x0000000104270e60 _PyEval_EvalFrameDefault + 24940
83  python3.10                          0x000000010426a364 _PyEval_Vector + 2036
84  python3.10                          0x0000000104186d10 method_vectorcall + 344
85  QtCore.abi3.so                      0x000000016a23e418 _ZNK8PyQtSlot4callEP7_objectS1_ + 156
86  QtCore.abi3.so                      0x000000016a23e22c _ZNK8PyQtSlot6invokeEPPvP7_objectS0_b + 296
87  QtCore.abi3.so                      0x000000016a23f200 _ZN13PyQtSlotProxy7unislotEPPv + 92
88  QtCore.abi3.so                      0x000000016a23f170 _ZN13PyQtSlotProxy11qt_metacallEN11QMetaObject4CallEiPPv + 68
89  QtCore                              0x000000016a54f1f4 _ZN11QMetaObject8activateEP7QObjectPKS_iPPv + 3144
90  QtGui                               0x000000016f4030a4 _ZN7QAction8activateENS_11ActionEventE + 368
91  QtCore                              0x000000016a547ed8 _ZN7QObject5eventEP6QEvent + 612
92  QtWidgets                           0x000000016fd98594 _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent + 272
93  QtWidgets                           0x000000016fd993e8 _ZN12QApplication6notifyEP7QObjectP6QEvent + 504
94  QtWidgets.abi3.so                   0x000000016ea67224 _ZN15sipQApplication6notifyEP7QObjectP6QEvent + 248
95  QtCore                              0x000000016a505474 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent + 292
96  QtCore                              0x000000016a5066fc _ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData + 1448
97  libqcocoa.dylib                     0x000000028a0eecb0 qt_plugin_instance + 54476
98  libqcocoa.dylib                     0x000000028a0efd74 qt_plugin_instance + 58768
99  CoreFoundation                      0x00000001853c263c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
100 CoreFoundation                      0x00000001853c25d0 __CFRunLoopDoSource0 + 176
101 CoreFoundation                      0x00000001853c2340 __CFRunLoopDoSources0 + 244
102 CoreFoundation                      0x00000001853c0f48 __CFRunLoopRun + 828
103 CoreFoundation                      0x00000001853c04b8 CFRunLoopRunSpecific + 612
104 HIToolbox                           0x000000018ec12df0 RunCurrentEventLoopInMode + 292
105 HIToolbox                           0x000000018ec12a80 ReceiveNextEventCommon + 220
106 HIToolbox                           0x000000018ec12984 _BlockUntilNextEventMatchingListInModeWithFilter + 76
107 AppKit                              0x00000001885e797c _DPSNextEvent + 636
108 AppKit                              0x00000001885e6b18 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716
109 AppKit                              0x00000001885daf7c -[NSApplication run] + 464
110 libqcocoa.dylib                     0x000000028a0ed8fc qt_plugin_instance + 49432
111 QtCore                              0x000000016a50f160 _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE + 540
112 QtCore                              0x000000016a505b00 _ZN16QCoreApplication4execEv + 112
113 QtWidgets.abi3.so                   0x000000016eb7039c _ZL22meth_QApplication_execP7_objectS0_ + 96
114 python3.10                          0x00000001041d6bbc cfunction_call + 208
115 python3.10                          0x000000010418359c _PyObject_MakeTpCall + 612
116 python3.10                          0x00000001042755d8 call_function + 676
117 python3.10                          0x0000000104270e60 _PyEval_EvalFrameDefault + 24940
118 python3.10                          0x000000010426a364 _PyEval_Vector + 2036
119 python3.10                          0x0000000104271658 _PyEval_EvalFrameDefault + 26980
120 python3.10                          0x000000010426a364 _PyEval_Vector + 2036
121 python3.10                          0x0000000104186c64 method_vectorcall + 172
122 python3.10                          0x0000000104275540 call_function + 524
123 python3.10                          0x0000000104270e60 _PyEval_EvalFrameDefault + 24940
124 python3.10                          0x000000010426a364 _PyEval_Vector + 2036
125 python3.10                          0x0000000104275540 call_function + 524
126 python3.10                          0x0000000104271430 _PyEval_EvalFrameDefault + 26428
127 python3.10                          0x000000010426a364 _PyEval_Vector + 2036
128 python3.10                          0x0000000104275540 call_function + 524
129 python3.10                          0x00000001042713bc _PyEval_EvalFrameDefault + 26312
130 python3.10                          0x000000010426a364 _PyEval_Vector + 2036
131 python3.10                          0x0000000104275540 call_function + 524
132 python3.10                          0x00000001042713bc _PyEval_EvalFrameDefault + 26312
133 python3.10                          0x000000010426a364 _PyEval_Vector + 2036
134 python3.10                          0x0000000104269b40 PyEval_EvalCode + 120
135 python3.10                          0x0000000104265760 builtin_exec + 836
136 python3.10                          0x00000001041d7880 cfunction_vectorcall_FASTCALL + 200
137 python3.10                          0x0000000104275540 call_function + 524
138 python3.10                          0x00000001042713bc _PyEval_EvalFrameDefault + 26312
139 python3.10                          0x000000010426a364 _PyEval_Vector + 2036
140 python3.10                          0x0000000104275540 call_function + 524
141 python3.10                          0x00000001042713bc _PyEval_EvalFrameDefault + 26312
142 python3.10                          0x000000010426a364 _PyEval_Vector + 2036
143 python3.10                          0x00000001042e8fc4 pymain_run_module + 272
144 python3.10                          0x00000001042e8604 Py_RunMain + 1584
145 python3.10                          0x00000001042e9a54 pymain_main + 1180
146 python3.10                          0x000000010412d31c main + 56
147 dyld                                0x0000000184f8bf28 start + 2236
```

**Comment by psobolewskiPhD** [comment]
I've been upgraded to macOS 14.2.1 and can no longer reproduce the issue.

**Comment by dalthviz** [comment]
Hi, just in case, checked on my macOS setup (Ventura 13.6.3 arm) launching from latest `main` (4a1b0b54) and I was unable to reproduce the crash (no crash when toggling 2D to 3D or showing the plugins dialog for me).

Napari installation details:

```

napari: 0.5.0a2.dev516+g4a1b0b54
Platform: macOS-13.6.3-arm64-arm-64bit
System: MacOS 13.6.3
Python: 3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:37:07) [Clang 15.0.7 ]
Qt: 6.6.1
PyQt6: 
NumPy: 1.26.2
SciPy: 1.11.4
Dask: 2023.11.0
VisPy: 0.14.1
magicgui: 0.8.0
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.2
npe2: 0.7.3

OpenGL:
  - GL version:  2.1 Metal - 83.1
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1312x848, scale 2.0

Settings path:
  - /Users/dalthviz/Library/Application Support/napari/napari-dev-arm_c26c98e387f96c964a1432a8a06da1b309926ebe/settings.yaml
Plugins:
  - napari: 0.5.0a2.dev516+g4a1b0b54 (81 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-svg: 0.1.10 (2 contributions)
```

**Comment by GenevieveBuckley** [comment]
I can't reproduce this on MacOS Sonoma 14.2.1 (23C71), for either the current napari release (0.4.19post1) or the current napari main branch ([commit 74b34a8](https://github.com/napari/napari/commit/74b34a8d33cc0b65172c5d9d25f459a27e329196)). I think we can close this issue now.

<details>
<summary>Details napari 0.4.19post1 (click to expand)</summary>

```
napari: 0.4.19.post1
Platform: macOS-14.2.1-arm64-arm-64bit
System: MacOS 14.2.1
Python: 3.11.8 | packaged by conda-forge | (main, Feb 16 2024, 20:49:36) [Clang 16.0.6 ]
Qt: 6.6.1
PyQt6: 
NumPy: 1.26.4
SciPy: 1.12.0
Dask: 2024.2.1
VisPy: 0.14.1
magicgui: 0.8.2
superqt: 0.6.2
in-n-out: 0.1.9
app-model: 0.2.4
npe2: 0.7.4

OpenGL:
- GL version: 2.1 Metal - 88
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1440x900, scale 2.0
- screen 2: resolution 1920x1080, scale 1.0

Settings path:
- /Users/genevieb/Library/Application Support/napari/napari-issue-6550_8199b6f363dd39dac8984b4afea48e24791d1418/settings.yaml
```

</details>

<details>
<summary>Details napari main branch (click to expand)</summary>

```
napari: 0.5.0a2.dev590+g74b34a8d
Platform: macOS-14.2.1-arm64-arm-64bit
System: MacOS 14.2.1
Python: 3.11.8 | packaged by conda-forge | (main, Feb 16 2024, 20:49:36) [Clang 16.0.6 ]
Qt: 6.6.1
PyQt6: 
NumPy: 1.26.4
SciPy: 1.12.0
Dask: 2024.2.1
VisPy: 0.14.1
magicgui: 0.8.2
superqt: 0.6.2
in-n-out: 0.1.9
app-model: 0.2.4
npe2: 0.7.4

OpenGL:
- GL version: 2.1 Metal - 88
- MAX_TEXTURE_SIZE: 16384
- GL_MAX_3D_TEXTURE_SIZE: 2048

Screens:
- screen 1: resolution 1440x900, scale 2.0
- screen 2: resolution 1920x1080, scale 1.0

Settings path:
- /Users/genevieb/Library/Application Support/napari/napari-issue-6550_8199b6f363dd39dac8984b4afea48e24791d1418/settings.yaml
```

</details>

**Comment by psobolewskiPhD** [comment]
Ran into this issue when trying to open RGB (astronaut) during community meeting on freshly pulled main after not working with napari for a few days on my laptop (just on our HPC)>
My env hadn't changed so I tried some git bisects to see if something in the repo caused it....and it brought me back to my freshly pulled main 👀 
...and in fact everything was working again.
So this doesn't seem fully resolved, but somehow intermittent. Never saw this using 6.6.0 for a while so I think there is a bug in qt 6.6.1 as noted above fixed in 6.6.2, plus also I found this issue https://bugreports.qt.io/browse/QTBUG-120469 which is similar fixed in 6.6.2.
So it may be worth flagging != on 6.6.1 at least on macOS.
